localstack/services/s3/validation.py

```diff
@@ -64,30 +76,52 @@ def validate_canned_acl(canned_acl: str) -> None:
     """
     Validate the canned ACL value, or raise an Exception
     """
-    if canned_acl and not is_canned_acl_bucket_valid(canned_acl):
+    if canned_acl and canned_acl not in VALID_CANNED_ACLS:
         ex = _create_invalid_argument_exc(None, "x-amz-acl", canned_acl)
         raise ex
 
 
-def validate_grantee_in_headers(grant: str, grantees: str) -> None:
+def parse_grants_in_headers(permission: Permission, grantees: str) -> Grants:
     splitted_grantees = [grantee.strip() for grantee in grantees.split(",")]
-    for grantee in splitted_grantees:
-        grantee_type, grantee_id = grantee.split("=")
+    grants = []
+    for seralized_grantee in splitted_grantees:
+        grantee_type, grantee_id = seralized_grantee.split("=")
         grantee_id = grantee_id.strip('"')
         if grantee_type not in ("uri", "id", "emailAddress"):
             ex = _create_invalid_argument_exc(
-                "Argument format not recognized", get_header_name(grant), grantee
+                "Argument format not recognized",
+                get_permission_header_name(permission),
+                seralized_grantee,
             )
             raise ex
-        elif grantee_type == "uri" and grantee_id not in s3_constants.VALID_ACL_PREDEFINED_GROUPS:
-            ex = _create_invalid_argument_exc("Invalid group uri", "uri", grantee_id)
-            raise ex
-        elif grantee_type == "id" and not is_valid_canonical_id(grantee_id):
-            ex = _create_invalid_argument_exc("Invalid id", "id", grantee_id)
-            raise ex
-        elif grantee_type == "emailAddress":
+        elif grantee_type == "uri":
+            if grantee_id not in s3_constants.VALID_ACL_PREDEFINED_GROUPS:
+                ex = _create_invalid_argument_exc("Invalid group uri", "uri", grantee_id)
+                raise ex
+            grantee = Grantee(
+                Type=GranteeType.Group,
+                URI=grantee_id,
+            )
+
+        elif grantee_type == "id":
+            if not is_valid_canonical_id(grantee_id):
+                ex = _create_invalid_argument_exc("Invalid id", "id", grantee_id)
+                raise ex
+            grantee = Grantee(
+                Type=GranteeType.CanonicalUser,
+                ID=grantee_id,
+                DisplayName="webfile",  # TODO: only in certain regions
```

Comment:
Yes, we should have a list of those regions in the S3 documentation, this would be great for parity! Right now we have the same behaviour as moto, but would be great to expand on it